src/server/ftpserver/options.rs

```diff
@@ -177,7 +178,7 @@ impl Default for SiteMd5 {
 
 /// The options for [Server.shutdown_indicator](crate::Server::shutdown_indicator) that allows users
 /// to specify the way in which a (graceful) shutdown of libunftp should happen.
-pub enum Shutdown<Signal: Future<Output = ()> + Send + Sync> {
+pub enum Shutdown<Signal: Future<Output = Duration> + Send + Sync> {
```

Comment:
I'm not really happy with the interface here, I may change it still.

current usage:

```
    let mut server = Server::with_authenticator(storage_backend, Arc::new(authenticator))
        .greeting("Welcome to unFTP")
        .passive_ports(start_port..end_port)
        .idle_session_timeout(idle_timeout)
        .logger(root_log.new(o!("lib" => "libunftp")))
        .passive_host(passive_host)
        .sitemd5(md5_setting)
        .shutdown_indicator(options::Shutdown::GracefulAcceptingConnections( async move {
            shutdown.recv().await.ok();
            info!(l, "Shutting down FTP server");
            Duration::from_secs(10)
        }))
        .metrics();
```